Applet/src/com/android/javacard/keymaster/KMKeymasterApplet.java

```diff
@@ -3876,31 +3947,35 @@ private static void encryptSecret(byte[] scratchPad) {
     data[SECRET] = KMByteBlob.instance(scratchPad, (short)0, len);
   }
 
-  private static void makeAuthData(byte[] scratchPad) {
-    short arrayLen = 2;
-    if (KMArray.cast(data[KEY_BLOB]).length() == 5) {
-      arrayLen = 3;
+  private static void makeAuthData(short version) {
+    short pos = 0;
+    short arrayLen = (version > 0) ? (short) 3 : (short) 2;
+    short maxArraySize = (version > 0) ? (short) ASYM_KEY_BLOB_SIZE : (short) 5;
```

Comment:
use a switch to decide the size based on version.

Reply:
Updated the code used switch case.